src/main/java/com/limechain/runtime/hostapi/StorageHostFunctions.java

```diff
@@ -28,11 +30,11 @@
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class StorageHostFunctions {
     private final HostApi hostApi;
-    private final KVRepository<String, Object> repository;
+    private final BlockTrieAccessor repository;
 
     private StorageHostFunctions(final HostApi hostApi) {
         this.hostApi = hostApi;
-        this.repository = SyncedState.getInstance().getRepository();
+        this.repository = AccessorHolder.getInstance().getBlockTrieAccessor();
```

Comment:
Do we assume that on every runtime initialization, the AccessHolder holds the trie accessor for the block required by the runtime?